enhancements/baremetal/baremetal.md

```diff
@@ -0,0 +1,283 @@
+
+The goal of this enhancement request is to provide context for all the changes
+that have already been merged towards making Baremetal IPI deployments a reality.
+All future Baremetal enhancement requests will refer back to this one to provide
+context.
+
+### Non-Goals
+
+Raising development PRs as a result of this enhancement request.
+
+## Proposal
+
+Every OpenShift based Baremetal IPI deployment will run a "metal3" pod on
+one Master Node. A "metal3" pod includes a container running BareMetal
+Operator(BMO) and several other supporting containers that work together.
+
+The BMO and other supporting containers together are able to discover a
+baremetal server in a pre-determined provisioning network, learn the
+HW attributes of the server and eventually boot it to make it available
+as a Machine within a MachineSet.
+
+The Machine API Operator (MAO) currently deploys the "metal3" pod only
+when the Platform type is `BareMetal` but the BaremetalHost CRD is exposed
+by the MAO as part of the release payload which is managed by the cluster
+version operator. The MAO is responsible for starting the BMO and the
+containers running the Ironic services and for providing these containers
+with their necessary configurations via env vars.
+
+The installer is responsible for kicking off a Baremetal IPI deployment
+with the right configuration.
+
+### User Stories
+
+With the addition of features described in this and other enhancements
+detailed in this current directory, OpenShift can be used to bring up
+a functioning cluster starting with a set of baremetal servers. As
+mentioned earlier, these enhancements rely on the Baremetal Operator (BMO)
+[1] running within the "metal3" pod to manage baremetal hosts. The BMO in
+turn relies on the Ironic service [3] to manage and provision baremetal
+servers.
+
+1. Will enable the user to deploy a control plane with 3 master nodes.
+2. Will enable the user to grow the cluster by dynamically adding worker
+nodes.
+3. Will enable the user to scale down the cluster by removing worker nodes.
+
+### Implementation Details/Notes/Constraints
+
+Baremetal IPI is integrated with OpenShift through the metal3.io [8] project.
+Metal3.io is a set of Kubernetes controllers that wrap the OpenStack Ironic
+project to provide Kubernetes native APIs for managing deployment and
+monitoring of physical hosts.
+
+The installer support for Baremetal IPI deployments is described in more detail
+in [7]. The installer runs on a special "provisioning host" that needs to be
+connected to both a "provisioning network" and an "external network". The
+provisioning network is a dedicated network used just for the purposes of
+configuring baremetal servers to be part of the cluster. The traffic on the
+provisioning network needs to be isolated from the traffic on the external
+network (hence 2 seperate networks.). The external network is used to carry
+cluster traffic which which includes cluster control plane traffic, application
```

Comment:
lift the network section from the install doc. It's nice and crisp.  Make a section for it here.

Reply:
for reference, https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md